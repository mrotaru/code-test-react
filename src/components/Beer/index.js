import React from 'react'

class Beer extends React.Component {
  constructor(props) {
    super(props)
    this.state = { showButton: false }
    this.onMouseEnter = this.onMouseEnter.bind(this)
    this.onMouseLeave = this.onMouseLeave.bind(this)
  }
  onMouseEnter () {
    this.setState({ showButton: true })
  }
  onMouseLeave () {
    this.setState({ showButton: false })
  }
  render() {
    const { beer, addToCart, inCart } = this.props
    const { showButton } = this.state
    return (
      <div onMouseEnter={this.onMouseEnter} onMouseLeave={this.onMouseLeave}>
        <span>{beer.name}</span>
        {inCart && <span>( {inCart} )</span>}
        {showButton &&
        <button onClick={() => {
          addToCart(beer)
        }}> + </button>}
      </div>
    )
  }
}

export default Beer