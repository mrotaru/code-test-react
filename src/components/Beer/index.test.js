import React from 'react'
import Beer from '.';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const beer = { name: 'Foo Beer'}
  const tree = renderer
    .create(<Beer beer={beer}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
