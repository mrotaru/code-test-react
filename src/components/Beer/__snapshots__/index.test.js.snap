// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span>
    Foo Beer
  </span>
</div>
`;
